File: Jadencase/jadencase.py
tweet = "How can mirrors be real if our eyes aren't real"

# Declaracion
def jadencase(text):
    #Separamos la frase en palabras
    words = text.split(" ")
    #Reemplazamos cada palabra con la version capitalizada
    for i, word in enumerate(words):
        words[i] = word.capitalize()
    #Unimos las cadenas de texto y devolvemos el resultado
    return " ".join(words)
    
def jadenCaseCopy(text):
    cadena = ""
    for i,letra in enumerate(text):
        if (i > 0 and text[i-1] == " "):
            cadena += letra.upper()
        else:
            cadena += letra
    return cadena


print(jadenCaseCopy(tweet))

File: Jadencase/jadencase.dart
final String text = "How can mirrors be real if our eyes aren't real";
main() {
  print(jadenCaseCopy(text));
}

//O(2n)
String jadenCase(String text) {
  List<String> words = text.split(' ');
  for (int i = 0; i < words.length; i++) {
    words[i] = "${words[i][0].toUpperCase()}${words[i].substring(1)}";
  }

  return words.join(" ");
}

//O(n)
String jadenCaseCopy(String text) {
  String string = "";
  for (int i = 0; i < text.length; i++) {
    string += i > 0 && text[i - 1] == " " ? text[i].toUpperCase() : text[i];
  }
  return string;
}
